src/vctrs.h

```diff
@@ -17,6 +35,8 @@ bool is_scalar(SEXP x);
 int equal_object(SEXP x, SEXP y, bool na_equal);
 bool equal_names(SEXP x, SEXP y);
 
+// Caller must ensure proper types and sizes. These functions are meant
```

Comment:
It's not clear to me what "proper types and sizes" are. Can you be more specific?

Reply:
This is now:

```c
// Caller must ensure proper SEXPTYPEs and lengths for `x` and `y`.
// These functions are meant to be used in loops. These must check
// bounds.
```